### PR TITLE
Refactor Voice module to use MapboxNavigationConsumer generic callback

### PR DESCRIPTION
**Pull request by Guardiola31337**
### Description

Refactors Voice module to use `MapboxNavigationConsumer` generic callback

Refs. https://github.com/mapbox/mapbox-navigation-android/pull/4125#discussion_r593478745

### Changelog

```
<changelog>Refactored Voice module to use `MapboxNavigationConsumer` generic callback.</changelog>
```

### CODE REVIEW

File: libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/model/SpeechVolume.kt [hunk 1]
```diff
@@ -2,11 +2,22 @@ package com.mapbox.navigation.ui.voice.model
...
     val level: Float
-)
+) {
+    init {
+        if (level !in MINIMUM_VOLUME_LEVEL..MAXIMUM_VOLUME_LEVEL) {
+            throw IllegalArgumentException(
+                "Level must be between [$MINIMUM_VOLUME_LEVEL..$MAXIMUM_VOLUME_LEVEL]"
+            )
+        }
+    }
```
Comment (Guardiola31337) [inline libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/model/SpeechVolume.kt:22]: Added this for consistency with other features https://github.com/mapbox/mapbox-navigation-android/pull/4109#discussion_r593485535

cc @abhishek1508

File: libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/model/SpeechVolume.kt [hunk 2]
```diff
@@ -2,11 +2,20 @@ package com.mapbox.navigation.ui.voice.model
...
-    @FloatRange(from = 0.0, to = 1.0)
+    @FloatRange(from = MINIMUM_VOLUME_LEVEL, to = MAXIMUM_VOLUME_LEVEL)
     val level: Float
-)
+) {
+    init {
+        require(level in MINIMUM_VOLUME_LEVEL..MAXIMUM_VOLUME_LEVEL) {
+            "Level must be between [$MINIMUM_VOLUME_LEVEL..$MAXIMUM_VOLUME_LEVEL]"
+        }
+    }
```
Comment (Guardiola31337) [inline libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/model/SpeechVolume.kt:20]: Added this for consistency with other features https://github.com/mapbox/mapbox-navigation-android/pull/4109#discussion_r593485535

cc @abhishek1508

File: libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/model/SpeechVolume.kt [hunk 3]
```diff
@@ -2,11 +2,20 @@ package com.mapbox.navigation.ui.voice.model
...
  */
 data class SpeechVolume(
-    @FloatRange(from = 0.0, to = 1.0)
+    @FloatRange(from = MINIMUM_VOLUME_LEVEL, to = MAXIMUM_VOLUME_LEVEL)
     val level: Float
-)
+) {
+    init {
+        require(level in MINIMUM_VOLUME_LEVEL..MAXIMUM_VOLUME_LEVEL) {
+            "Level must be between [$MINIMUM_VOLUME_LEVEL..$MAXIMUM_VOLUME_LEVEL]"
```
Comment (abhishek1508) [inline libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/model/SpeechVolume.kt:18]:
```suggestion
            "Volume level must be between [$MINIMUM_VOLUME_LEVEL..$MAXIMUM_VOLUME_LEVEL]"
```

**Review by abhishek1508 — APPROVED** [review]
LGTM. Suggested a NIT

**Pull request merged**